butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/pom/PomRemoveDependencyExclusion.java

```diff
@@ -0,0 +1,165 @@
+        TransformationUtility.checkForBlankString("exclusionGroupId", exclusionGroupId);
+        this.exclusion.setGroupId(exclusionGroupId);
+        return this;
+    }
+
+    public PomRemoveDependencyExclusion setExclusionArtifactId(String exclusionArtifactId) {
+        TransformationUtility.checkForBlankString("exclusionArtifactId", exclusionArtifactId);
+        this.exclusion.setArtifactId(exclusionArtifactId);
+        return this;
+    }
+
+
+
+    @Override
+    public PomRemoveDependencyExclusion failIfNotPresent() {
+        ifNotPresent = ChangeOrRemoveElement.IfNotPresent.Fail;
+        return this;
+    }
+
+    @Override
+    public PomRemoveDependencyExclusion warnIfNotPresent() {
+        ifNotPresent = ChangeOrRemoveElement.IfNotPresent.Warn;
+        return this;
+    }
+
+    @Override
+    public PomRemoveDependencyExclusion noOpIfNotPresent() {
+        ifNotPresent = ChangeOrRemoveElement.IfNotPresent.NoOp;
+        return this;
+    }
+
+    public Exclusion getExclusion() {
+        return this.exclusion;
+    }
+
+    @Override
+    public String getDescription() {
+        return String.format(DESCRIPTION, exclusion.getGroupId(), exclusion.getArtifactId(), groupId,
+                artifactId, getRelativePath());
+    }
+
+    @Override
+    protected TOExecutionResult pomExecution(String relativePomFile, Model model) {
+        TOExecutionResult result = null;
+        String details;
+
+        Dependency dependency = getDependency(model, groupId, artifactId);
+        if(dependency == null) {
+            dependency = getDependencyFromDependencyManagement(model);
+        }
+        if (dependency != null) {
+
+            List<Exclusion> exclusions = dependency.getExclusions();
+
+            if(exclusions != null && exclusions.size() > 0) {
+                removeExclusionFromDependency(dependency, exclusions);
+                details = String.format("Exclusion %s:%s has been removed from dependency %s:%s in POM file %s",
+                        exclusion.getGroupId(), exclusion.getArtifactId(), groupId, artifactId, relativePomFile);
+                result = TOExecutionResult.success(this, details);
+            } else {
+                details = String.format("Exclusion %s:%s not present in dependency %s:%s in POM file %s",
+                        exclusion.getGroupId(), exclusion.getArtifactId(), groupId, artifactId, relativePomFile);
+                result = getTOExecutionResult(details);
+            }
+        } else {
+            details = String.format("Exclusion %s:%s has not been removed as dependency %s:%s in POM file %s because dependency is not present",
+                    exclusion.getGroupId(), exclusion.getArtifactId(), groupId, artifactId, relativePomFile);
+            result = getTOExecutionResult(details);
+        }
+        return result;
+    }
+
+    private TOExecutionResult getTOExecutionResult(String details) {
```

Comment:
This method name is confusing. It implies that a result object can be returned in any circumstance. However, by reading the code that calls this method, it looks like that is applicable only when the result is different than success.

Please rename it to `getUnsuccessfulResult`.